src/lib/rest/rest.cpp

```diff
@@ -464,17 +466,54 @@ static void requestCompleted
   MHD_RequestTerminationCode  toe
 )
 {
-  ConnectionInfo* ciP      = (ConnectionInfo*) *con_cls;
+  ConnectionInfo*  ciP      = (ConnectionInfo*) *con_cls;
+  struct timespec  reqEndTime;
 
   if ((ciP->payload != NULL) && (ciP->payload != static_buffer))
   {
     free(ciP->payload);
   }
 
-  delete(ciP);
   *con_cls = NULL;
 
   LM_TRANSACTION_END();  // Incoming REST request ends
+
+  if (reqTimeStatistics)
+  {
+    clock_gettime(CLOCK_REALTIME, &reqEndTime);
+    clock_difftime(&reqEndTime, &ciP->reqStartTime, &threadLastTimeStat.reqTime);
+  }  
+
+  delete(ciP);
+
+  //
+  // Statistics
+  //
+  // Flush this requests timing measures onto a global var to be read by "GET /statistics".
+  // Also, increment the accumulated measures.
+  if (reqTimeStatistics)
+  {
+    timeStatSemTake(__FUNCTION__, "updating statistics");
+
+    memcpy(&lastTimeStat, &threadLastTimeStat, sizeof(lastTimeStat));
+
+    // "Fix" mongoBackendTime
```

Comment:
I'd suggest a more elaborate description (we can forget what this "fix" mean in a couple of months and think it was due to some unknown bug or something like that), e.g.:

```
// Substract times waiting at mongo driver operation (in mongo[Read|Write|Command]WaitTime counters) so mongoBackendTime contains at the end the time passed in our logic, i.e. a kind of "self-time" for mongoBackend
```